core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1223,6 +1223,12 @@ class UnifiedLog(@volatile var logStartOffset: Long,
         s"but we only have log segments starting from offset: $logStartOffset.")
   }
 
+  private def checkLocalLogStartOffset(offset: Long): Unit = {
```

Comment:
It seems reading records between [logStartOffset, localLogStartOffset] is dangerous since the segment won't be in local-disk. That is a bit chaos to me as `UnifiedLog` presents a unified view of local and tiered log segment (https://github.com/apache/kafka/blob/fccd7fec666d6570758e0b7891771099240ceee8/core/src/main/scala/kafka/log/UnifiedLog.scala#L59). The check looks like a limit that we can't "view" data from tiered log segment.

Reply:
Agree on this. The `checkLocalLogStartOffset` is used only in the `convertToOffsetMetadataOrThrow` method which reads from local-disk.